picard/ui/searchdialog.py

```diff
@@ -0,0 +1,466 @@
+            self.table.setItem(row, 3, table_item(track.get("album", "")))
+            self.table.setItem(row, 4, table_item(track.get("date", "")))
+            self.table.setItem(row, 5, table_item(track.get("country", "")))
+            self.table.setItem(row, 6, table_item(track.get("releasetype", "")))
+
+    def parse_tracks_from_xml(self, tracks_xml):
+        """Extracts track information from XmlNode objects and stores that into Metadata objects.
+
+        Args:
+            tracks_xml -- list of XmlNode objects
+        """
+        for node in tracks_xml:
+            if "release_list" in node.children and "release" in node.release_list[0].children:
+                for rel_node in node.release_list[0].release:
+                    track = Metadata()
+                    recording_to_metadata(node, track)
+                    release_to_metadata(rel_node, track)
+                    rg_node = rel_node.release_group[0]
+                    release_group_to_metadata(rg_node, track)
+                    if "release_event_list" in rel_node.children:
+                        # Extract contries list from `release_event_list` element
+                        # Don't use `country` element as it contains information of a single release
+                        # event and is basically for backward compatibility.
+                        country = []
+                        for re in rel_node.release_event_list[0].release_event:
+                            try:
+                                country.append(
+                                        re.area[0].iso_3166_1_code_list[0].iso_3166_1_code[0].text)
+                            except AttributeError:
+                                pass
+                        track["country"] = ", ".join(country)
+                    self.search_results.append((track, node))
+            else:
+                # This handles the case when no release is associated with a track
+                # i.e. the track is a NAT
+                track = Metadata()
+                recording_to_metadata(node, track)
+                track["album"] = _("Standalone Recording")
+                self.search_results.append((track, node))
+
+    def load_selection(self, row):
+        """Loads album corresponding to selected track.
+        If the search is performed for a file, also associates the file to
+        corresponding track in the album.
+        """
+        track, node = self.search_results[row]
+        if track.get("musicbrainz_albumid"):
+        # The track is not an NAT
+            self.tagger.get_release_group_by_id(track["musicbrainz_releasegroupid"]).loaded_albums.add(
+                    track["musicbrainz_albumid"])
+            if self.file_:
+            # Search is performed for a file
+            # Have to move that file from its existing album to the new one
+                if type(self.file_.parent).__name__ == "Track":
```

Comment:
You should use `if isinstance(self.file_.parent, Track)` (with the appropriate imports of course) instead.